lib/viewcomponents/app.jsx

```diff
@@ -1,28 +1,33 @@
-var React = require('react');
+var CreditsPanel = require('./gui/panels/credits.jsx');
+var GCLPanel = require('./gui/panels/gcl.jsx');
 var Graph = require('./graph.jsx');
+var React = require('react');
+var Toolbar = require('./gui/toolbar.jsx');
 
 /**
  * App is the top-level component. It fills the whole body of the page.
  */
-var App = React.createClass({
+ var App = React.createClass({
 
   render: function() {
-    var gclEditorModel = this.props.gclEditorModel;
     return (
       <div id='app'>
-        <div id='model_editor'>
-          <textarea onChange={this._onGCLChange} defaultValue={gclEditorModel.get('text')}>
-          </textarea>
-        </div>
-        <Graph model={gclEditorModel.get('graph')} />
+        <Toolbar tools={{
+          gcl: {
+            icon: '/images/icons/gcl.svg',
+            title: 'Content Code Editor',
+            panelContent: <GCLPanel model={this.props.gclEditorModel} />
+          },
+
+          credits: {
+            icon: '/images/icons/about.svg',
+            title: 'Credits',
+            panelContent: <CreditsPanel />
```

Comment:
I think that this is appropriate, considering that you want these panels to have a uniform frame around them, dictated by the toolbar. The solution that I asked my colleague about would be appropriate if that weren't the case.